pkg/moov/bank_account.go

```diff
@@ -80,15 +82,59 @@ type MX struct {
 	AuthorizationCode string `json:"authorizationCode"`
 }
 
+type CreateBankAccountType callArg
+
+func WithBankAccount(bankAccount BankAccount) CreateBankAccountType {
+	return callBuilderFn((func(call *callBuilder) error {
+		bankAccountJSON, err := json.Marshal(bankAccount)
+		if err != nil {
+			return err
+		}
+		call.body = bytes.NewReader(bankAccountJSON)
+		return nil
+	}))
+}
+
+func WithPlaidLink(plaidLink PlaidLink) CreateBankAccountType {
+	return callBuilderFn((func(call *callBuilder) error {
+		bankAccountJSON, err := json.Marshal(plaidLink)
+		if err != nil {
+			return err
+		}
+		call.body = bytes.NewReader(bankAccountJSON)
+		return nil
+	}))
+}
+
+func WithPlaid(plaid Plaid) CreateBankAccountType {
+	return callBuilderFn((func(call *callBuilder) error {
+		bankAccountJSON, err := json.Marshal(plaid)
+		if err != nil {
+			return err
+		}
+		call.body = bytes.NewReader(bankAccountJSON)
+		return nil
+	}))
+}
+
+func WithMX(mx MX) CreateBankAccountType {
```

Comment:
I don't think that the WithMX will work as you intended.

From what I can tell we expect the JSON to the server to be in the format

```
"mx": {
  "authorizationCode": "test code"
}
```

the current implementation will output just
```
{
"authorizationCode": "testing"
}

without the top level MX key. same issue with the WithPlaidLink and WithPlaid helpers.

Reply:
I think you could update this to do it like

```go
func WithMX(mx MX) CreateBankAccountType {
	return callBuilderFn((func(call *callBuilder) error {
		// need to use a map to add the top level "mx" json key. expected format is:
		//	{
		// 		"mx": {
		// 			"authorizationCode": "testing"
		// 		}
		// 	}
		//
		mxMap := map[string]MX{
			"mx": mx,
		}
		bankAccountJSON, err := json.Marshal(mxMap)
		if err != nil {
			return err
		}
		call.body = bytes.NewReader(bankAccountJSON)
		return nil
	}))
}
```

Reply:
i ended up coding these things up if you want to see them - https://github.com/moovfinancial/moov-go/pull/80

Reply:
Did we resolve this?